### PR TITLE
lesson_03

### PR DESCRIPTION
**Pull request by akpoik**
@gradopaw

### CODE REVIEW

**Comment by akpoik** [comment]
@gradopaw

**Comment by akpoik** [comment]
@gradopaw

**Comment by akpoik** [comment]
@gradopaw

File: students/kulon_aleksander/lesson_03/exponentiation.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    if n > 1:
+        n = n - 1
+        a = a * power(a, n)
+        return a
+    else:
+        return a
+
+
+def main():
+    aa = float(input())
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/exponentiation.py]: When you moved this part to ```main()``` function you don't need to care about shadowing variables from outer scope - now the ```aa``` and ```nn``` variables are local variables used within the function. Of course now is ok :)

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    print("h,? - help")
+    print("q - QUIT")
+
+
+def adding(var_1, var_2):
+    print("ADDING")
+    return var_1 + var_2
+
+
+def substract(var_1, var_2):
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: substract or subtract? Typo :)

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    print("SUBTRACT")
+    return (var_1 - var_2)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: remove redundant parentheses. ```return var_1 - var_2``` is enough.

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def multiply(var_1, var_2):
+    print("MULTIPLY")
+    return (var_1 * var_2)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: remove redundant parentheses. return var_1 * var_2 is enough.

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def divide(var_1, var_2):
+    print("DIVIDE")
+    return (var_1 / var_2)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: remove redundant parentheses. return var_1 / var_2 is enough.

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def power(var_1, var_2):
+    print("POWER")
+    return (var_1 ** var_2)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: remove redundant parentheses. return var_1 ** var_2 is enough.

File: students/kulon_aleksander/lesson_03/calculator.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+            elif option == "s":
+                result = substract(var_1, var_2)
+            elif option == "m":
+                result = multiply(var_1, var_2)
+            elif option == "d":
+                result = divide(var_1, var_2)
+            elif option == "p":
+                result = power(var_1, var_2)
+
+            print("Result:", result)
```
Comment (gradopaw) [inline students/kulon_aleksander/lesson_03/calculator.py]: What in the case when ```result``` variable will be no assignment? What should happen when user will input 'k' value? Maybe you should add else statement in line 69  and inform the user that inserted the wrong value and let him to input one more time proper value or just finish the program.

**Comment by akpoik** [comment]
@gradopaw

**Pull request merged**